src/app/components/Modals/TransitionModal/StageStepping.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+  Button,
+  DialogBody,
+  DialogFooter,
+  DialogHeader,
+  Loader,
+  Text,
+} from "@babylonlabs-io/bbn-core-ui";
+import { twMerge } from "tailwind-merge";
+
+import { Tick } from "./Tick";
+
+const StepContent = [
+  "Step 1: Consent to slashing",
+  "Step 2: Consent to slashing during unbonding ",
+  "Step 3: BTC-BBN address binding for receiving staking rewards",
+  "Step 4: Staking transaction registration",
+];
+
+interface StageSteppingProps {
+  onClose: () => void;
+  onSign: () => void;
+  step: number;
+  awaitingResponse?: boolean;
+}
+
```

Comment:
```suggestion
const Step = ({ active, completed, current, content }: { active: boolean; completed: boolean; current: number; content: string; }) => (
         <div
              key={content}
              className={twMerge(
                "p-4 flex flex-row items-center justify-start gap-3 rounded border border-primary-dark/20 bg-secondary-contrast self-stretch",
                !active && "opacity-25",
              )}
            >
              {completed ? (
                <Tick />
              ) : (
                <div className="rounded-full bg-secondary-main flex h-10 w-10 items-center justify-center">
                  <Text variant="body1" className="text-secondary-contrast">
                    {current}
                  </Text>
                </div>
              )}
              <Text variant="body1" className="text-primary-dark">
                {content}
              </Text>
            </div>
)
```

Reply:
My two cents: I typically avoid having too many state variables in a component because the combinations of these states grow exponentially, often leading to messy and complex code.